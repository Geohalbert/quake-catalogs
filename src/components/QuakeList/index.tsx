import React, { memo } from 'react';
import { Link } from 'react-router-dom';

import styles from './styles.scss';

type Props = {
  list: Array<{ id: string; name: string }>;
};

export default memo(({ list }: Props) => (
  <div className={styles.QuakeList}>
    <h4>Quake List</h4>
    <ul>
      {list.map(({ id, name }) => (
        <li key={id}>
          <Link to={`/QuakeInfo/${id}`}>{name}</Link>
        </li>
      ))}
    </ul>
  </div>
));
