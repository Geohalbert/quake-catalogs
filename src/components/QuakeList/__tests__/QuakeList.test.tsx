import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import QuakeList from '../index';

describe('<QuakeList />', () => {
  it('renders', () => {
    const mockData = [{ id: '1', name: 'Welly' }];
    const tree = renderer
      .create(
        <MemoryRouter>
          <QuakeList list={mockData} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
