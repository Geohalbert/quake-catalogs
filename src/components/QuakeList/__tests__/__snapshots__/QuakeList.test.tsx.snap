// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuakeList /> renders 1`] = `
<div
  className="QuakeList"
>
  <h4>
    Quake List
  </h4>
  <ul>
    <li>
      <a
        href="/QuakeInfo/1"
        onClick={[Function]}
      >
        Welly
      </a>
    </li>
  </ul>
</div>
`;
